WalletWasabi.Fluent/ViewModels/CoinJoinProfiles/CoinJoinProfilesViewModel.cs

```diff
@@ -75,22 +75,26 @@ private async Task OnManualSetupAsync()
 		}
 	}
 
-	private void OnNext(KeyManager keyManager, bool isNewWallet)
+	private void OnNext(IWalletModel walletModel, bool isNewWallet)
 	{
 		var selected = SelectedProfile ?? SelectedManualProfile ?? Profiles.First();
 
-		keyManager.RedCoinIsolation = selected.RedCoinIsolation;
-		keyManager.SetAnonScoreTarget(selected.AnonScoreTarget, toFile: false);
-		keyManager.SetFeeRateMedianTimeFrame(selected.FeeRateMedianTimeFrameHours, toFile: false);
-		keyManager.IsCoinjoinProfileSelected = true;
+		using (walletModel.Settings.BatchChanges())
```

Comment:
This is something that now doesn't case problem but later it can.

So when you create a wallet it is a must to only call KeyManager.ToFile() when the user ended with the wallet creation flow. (So it should only happen only on the last page). Because Imagine you have P1 P2 P3. You call ToSave on P1, the user changes their mind on P2 and quits the wallet creation flow. But since the file was created, the next time when they open the software, the wallet will be in the wallet list.

So it is not a problem now because ToFile is called on LastPage-1, but it can be a problem when there will be a new page between the current LastPage-1 and LastPage.
Long story short... ToFile shouldn't be called here.